core/controlplane/config/templates/cloud-config-worker

```diff
@@ -845,3 +860,414 @@ write_files:
+
+      # Load and unload nvidia-drm module
+      ACTION=="add" DEVPATH=="/module/nvidia" SUBSYSTEM=="module" RUN+="/opt/nvidia/current/bin/nvidia-insmod.sh nvidia-drm.ko"
+      ACTION=="remove" DEVPATH=="/module/nvidia" SUBSYSTEM=="module" RUN+="/usr/sbin/rmmod nvidia-drm"
+
+      # Load and unload nvidia-uvm module
+      ACTION=="add" DEVPATH=="/module/nvidia" SUBSYSTEM=="module" RUN+="/opt/nvidia/current/bin/nvidia-insmod.sh nvidia-uvm.ko"
+      ACTION=="remove" DEVPATH=="/module/nvidia" SUBSYSTEM=="module" RUN+="/usr/sbin/rmmod -r nvidia-uvm"
+
+      # This will create the device nvidia device nodes
+      ACTION=="add" DEVPATH=="/module/nvidia" SUBSYSTEM=="module" RUN+="/opt/nvidia/current/bin/nvidia-smi"
+
+      # Create the device node for the nvidia-uvm module
+      ACTION=="add" DEVPATH=="/module/nvidia_uvm" SUBSYSTEM=="module" RUN+="/opt/nvidia/current/bin/create-uvm-dev-node.sh"
+
+  - path: /opt/nvidia-build/_container_build.sh
+    owner: root:root
+    permissions: 0755
+    content: |
+      #!/bin/sh
+
+      # Default: use binary packages instead of building everything from source
+      EMERGE_SOURCE_FLAGS=gK
+      while :; do
+        case $1 in
+          --emerge-sources)
+            EMERGE_SOURCE_FLAGS=
+            ;;
+          *)
+            break
+        esac
+        shift
+      done
+
+
+      VERSION=$1
+      echo Building ${VERSION}
+
+      function finish {
+        cat /nvidia_installers/NVIDIA-Linux-x86_64-${VERSION}/nvidia-installer.log
+      }
+
+      set -e
+      trap finish exit
+
+      emerge-gitclone
+      . /usr/share/coreos/release
+      git -C /var/lib/portage/coreos-overlay checkout build-${COREOS_RELEASE_VERSION%%.*}
+      emerge -${EMERGE_SOURCE_FLAGS}q --jobs 4 --load-average 4 coreos-sources
+
+      cd /usr/src/linux
+      cp /lib/modules/*-coreos*/build/.config .config
+
+      make olddefconfig
+      make modules_prepare
+
+      cd /nvidia_installers/NVIDIA-Linux-x86_64-${VERSION}
+      ./nvidia-installer -s -n --kernel-source-path=/usr/src/linux \
+        --no-check-for-alternate-installs --no-opengl-files \
+        --kernel-install-path=${PWD} --log-file-name=${PWD}/nvidia-installer.log
+
+  - path: /opt/nvidia-build/_export.sh
+    owner: root:root
+    permissions: 0755
+    content: |
+      #!/bin/sh
+
+      set -e
+
+      ARTIFACT_DIR=$1
+      VERSION=$2
+      COMBINED_VERSION=$3
+
+      TOOLS="nvidia-debugdump nvidia-cuda-mps-control nvidia-xconfig nvidia-modprobe nvidia-smi nvidia-cuda-mps-server
+      nvidia-persistenced nvidia-settings"
+
+      # Create archives with no paths
+      tar -C ${ARTIFACT_DIR} -cvj $(basename -a ${ARTIFACT_DIR}/*.so.*) > libraries-${VERSION}.tar.bz2
+      tar -C ${ARTIFACT_DIR} -cvj ${TOOLS} > tools-${VERSION}.tar.bz2
+      tar -C ${ARTIFACT_DIR}/kernel -cvj $(basename -a ${ARTIFACT_DIR}/kernel/*.ko) > modules-${COMBINED_VERSION}.tar.bz2
+
+  - path: /opt/nvidia-build/build.sh
```

Comment:
> I believe this duration would be acceptable for many users because gpu node pool usually doesn't need to scale so quickly like normal nodepool which hosts service pods.

I completely agree with you here 👍 
The local build feature could be an extra thing we may or may not add in the future.